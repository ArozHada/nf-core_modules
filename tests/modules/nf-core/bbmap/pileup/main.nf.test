nextflow_process {

    name "Test Process BBMAP_PILEUP"
    script "modules/nf-core/bbmap/pileup/main.nf"
    process "BBMAP_PILEUP"
    config "tests/modules/nf-core/bbmap/pileup/nextflow.config"
    tag "bbmap_pileup"

    test("Sarscov2 Illumina PE BAM") {

        when {
            params {
                outdir = "$outputDir"
            }
            process {
                """
                input[0] = [
                            [ id:'test', single_end:false ], // meta map
                            file(params.test_data['sarscov2']['illumina']['test_paired_end_bam'], checkIfExists: true)
                           ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(path(params.outdir).list()).match() }
            )
        }

    }

}
