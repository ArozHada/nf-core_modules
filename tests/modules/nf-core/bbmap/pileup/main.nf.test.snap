{
    "Sarscov2 Illumina PE BAM": {
        "content": [
            [
                "test.coverage.hist.txt:md5,96915920ef42ddc9483457dd4585a088",
                "test.coverage.stats.txt:md5,c3fc9d0681589b69e3301ca3cb27b7a4",
                "versions.yml:md5,e293e3ed6387ce89f3443050f36fb079"
            ]
        ],
        "timestamp": "2023-04-08T14:34:34+0000"
    }
}